test/cpp_extensions/open_registration_extension/torch_openreg/csrc/runtime/OpenRegHooks.h

```diff
@@ -30,14 +66,24 @@ struct OpenRegHooksInterface : public at::PrivateUse1HooksInterface {
     return attr.type == orMemoryTypeHost;
   }
 
-  const at::Generator& getDefaultGenerator(
-      c10::DeviceIndex device_index) const override {
+  at::Device getDeviceFromPtr(void* data) const override {
+    orPointerAttributes attr{};
+    auto err = orPointerGetAttributes(&attr, data);
+    if (err == orSuccess && attr.type == orMemoryTypeDevice) {
+      return at::Device(at::DeviceType::PrivateUse1, static_cast<int>(attr.device));
+    }
+    // Fallback to CPU device for host/unmanaged pointers
+    return at::Device(at::DeviceType::CPU, 0);
```

Comment:
It would be better to raise error to user if the ptr passed in is not within any device memory range.